Simplified/OPDS/NYPLOPDSFeedFetcher.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  NYPLOPDSFeedFetcher.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2022-01-09.
+//  Copyright © 2022 NYPL. All rights reserved.
+//
+
+import Foundation
+
+@objcMembers class NYPLOPDSFeedFetcher: NSObject {
+  // The the maximum attempts of books fetching allowed until we receive supported books.
+  private static let fetchSupportedBooksRetryThreshold = 15
+  
+  ///   Fetch catalog feed for given URL.
+  ///   If the returned feed does not contain supported books, it will call itself recursively
+  ///   until supported books are received or retry threshold reached.
+  ///
+  ///   - Parameter url: URL for fetching the catalog feed
+  ///   - Parameter retryCount: Current number of attempts to fetch supported books, use 0 as default value
+  ///   - Parameter completion: Always invoked at the end no matter what,
+  ///   providing an ungrouped feed object in case of success and nil otherwise.
+  class func fetchCatalogUngroupedFeed(url: URL,
+                                       retryCount: Int,
+                                       completion: @escaping (_ feed: NYPLCatalogUngroupedFeed?) -> Void) {
+    
+    if (retryCount >= fetchSupportedBooksRetryThreshold) {
+      Log.info(#function, "Retry threshold reached while fetching Catalog feed.")
+      DispatchQueue.global(qos: .userInitiated).async {
```

Comment:
The reason I did this is because this function can possibly recursively call itself multiple times, which can be a time-consuming process. Running the completion handler in `DispatchQueue.global(qos: .userInitiated)` gives it a higher priority and complete the fetch faster. I am not 100% sure on this, so feel free to make any suggestions.

Reply:
Sure, I will call it directly then.

Reply:
Done